### PR TITLE
Fix provisioning and sidekiq

### LINKED ISSUE #57: Upstart script broken
`Foreman`'s writing of upstart script doesn't seem to really make sense as we have it configured now. Should definitely look into that.

### PR DESCRIPTION
**Pull request by hut8**
Resolves #57 and #60

### CODE REVIEW

File: app/controllers/main_controller.rb [hunk 1]
```diff
@@ -111,6 +111,10 @@ def indexed_packages
   private
 
   def redirect_to_https
-    redirect_to protocol: 'https://' unless request.ssl? || request.local? || Rails.env.development?
+    redirect_to protocol: 'https://' unless request.ssl? || can_skip_https
+  end
+
+  def can_skip_https
```
Comment (joshjordan) [inline app/controllers/main_controller.rb]: Picky rubyism: boolean method should end in question mark: `can_skip_https?`

File: infrastructure/nginx-site.conf.j2 [hunk 2]
```diff
@@ -6,13 +7,14 @@ server {
...
     ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
+    {% endif %}
     
     server_name {{ inventory_hostname }};
     root /home/{{ app_name }}/rails-apps/{{ app_name }}/current/public;
     rails_env {{ rails_env }};
     
     passenger_enabled on;
-    passenger_min_instances 8;
+    passenger_min_instances 5;
```
Comment (joshjordan) [inline infrastructure/nginx-site.conf.j2]: How did you settle on 5?

**Comment by joshjordan** [comment]
Only concern is needing to make infrastructure changes when the Procfile changes, which is not a real concern at all. :+1:

File: app/controllers/main_controller.rb (hunk 1 above)
Comment (hut8) [inline app/controllers/main_controller.rb]: Fuck, you're absolutely right. Thanks.

File: infrastructure/nginx-site.conf.j2 (hunk 2 above)
Comment (hut8) [inline infrastructure/nginx-site.conf.j2]: 6, 7 and 8 were getting hit at most once or twice in the 4 day span I was looking at. 5 only had a few more, but better too high than too low?

Comment (hut8) [inline infrastructure/nginx-site.conf.j2]: Actually since this is min_instances, 3 might be better.

Comment (joshjordan) [inline infrastructure/nginx-site.conf.j2]: I like 3. I also like setting this to number of cores on the box. I suspect
that is 4.

Comment (hut8) [inline infrastructure/nginx-site.conf.j2]: You're very optimistic. It's actually 2.

Comment (joshjordan) [inline infrastructure/nginx-site.conf.j2]: Haha. I would just set it to 2 and see how it behaves. I am fine with any
number on [2,5] though.

**Pull request merged**